ios/RNCWKSchemeHandler.m

```diff
@@ -60,19 +84,22 @@ - (void)webView:(WKWebView *)webView startURLSchemeTask:(id <WKURLSchemeTask>)ur
 
   // Send off to React Native.
   [self.delegate handleUrlSchemeRequest:req];
-
 }
 
 - (void)handleUrlSchemeResponse:(NSDictionary *)resp
 {
   // Grab the task we want to complete.
   NSString *requestId = [resp objectForKey:@"requestId"];
-  id<WKURLSchemeTask> urlSchemeTask = [self.urlSchemeRequestTasks objectForKey:requestId];
+  id<WKURLSchemeTask> urlSchemeTask = [self getSchemeTaskForID:requestId];
```

Comment:
Is this guaranteed to happen on the main thread? Are reads thread safe?

Reply:
From Apple's docs on the matter it appears it is safe as long as it is not a mutating action. 
> In general, the collection classes (for example, NSMutableArray, NSMutableDictionary) are not thread-safe when mutations are concerned.
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Multithreading/ThreadSafetySummary/ThreadSafetySummary.html